src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -314,6 +290,110 @@ VectorDims LinearIR::get_master_shape() const {
+}
+
+LinearIR::exprIt LinearIR::replace_node(const std::shared_ptr<ov::Node>& new_node, const std::vector<ExpressionPtr>& old_exprs,
+                                        const std::vector<size_t>& loop_ids, const constExprIt& place) {
+    OPENVINO_ASSERT(!old_exprs.empty(), "Failed to replace node: there are no old expressions for replacing");
+     OPENVINO_ASSERT(new_node->get_output_size() == old_exprs.back()->get_output_count(),
+                    "Failed to replace node: node output port count is not equal to output count of last old expression");
+
+    std::vector<PortConnectorPtr> new_inputs(new_node->get_input_size());
+    for (size_t i = 0; i < new_node->get_input_size(); ++i) {
+        const auto& source = new_node->get_input_source_output(i);
+        new_inputs[i] = get_expr_by_node(source.get_node_shared_ptr())->get_output_port_connector(source.get_index());
+    }
+
+    auto is_old_expr = [&old_exprs](const ExpressionPtr& expr) {
+        return std::find(old_exprs.cbegin(), old_exprs.cend(), expr) != old_exprs.cend();
+    };
+    auto is_input_source = [&new_inputs](const ExpressionPort& source) {
+        return std::any_of(new_inputs.cbegin(), new_inputs.cend(), [&source](const PortConnectorPtr& input) { return input->get_source() == source; });
+    };
+
+    // Validate removable expressions - they must be a `sequence`
+    const auto last_old_expr = old_exprs.back();
+    std::vector<std::set<ExpressionPort>> consumers(last_old_expr->get_output_count());
+    for (const auto& old_expr : old_exprs) {
+        for (const auto& input : old_expr->get_input_port_connectors()) {
+            const auto& source = input->get_source();
+            OPENVINO_ASSERT(is_old_expr(source.get_expr()) || is_input_source(source),
+                            "Failed to replace nodes: not all output ports of existing expressions will be connected!");
+        }
+        for (const auto& port_connector : old_expr->get_output_port_connectors()) {
+            const auto old_expr_consumers = port_connector->get_consumers();
+            if (old_expr != last_old_expr) {
+                for (const auto& consumer : old_expr_consumers) {
+                    OPENVINO_ASSERT(is_old_expr(consumer.get_expr()),
+                                    "Failed to replace nodes: not all input ports of existing expressions will be connected!");
+                }
+            } else {
+                consumers[port_connector->get_source().get_index()] = old_expr_consumers;
+            }
+        }
+    }
+    for (size_t i = 0; i < new_node->get_output_size(); ++i)
+        snippets::lowered::PortDescriptorUtils::set_port_descriptor_ptr(new_node->output(i), last_old_expr->get_output_port_descriptor(0)->clone());
+
+    // Notes:
+    // - we cannot automatically updated loop ports in node insertion since there are old_exprs
+    // - we should update loop ports firstly and onlyt after that - remove old exprs from linearIR
```

Comment:
Typo
```suggestion
    // Notes:
    // - we cannot automatically update loop ports in node insertion since there are old_exprs
    // - we should update loop ports firstly and only after that - remove old exprs from linearIR
```

Reply:
Done, thanks